[BUGFIX] 매칭내역 조회 API 수정

ujiiin opened the pull request.

ujiiin committed:
[BUGFIX] guest, guest-user dao파일 구분 #110

ujiiin committed:
[BUGFIX] 매칭내역 조회 타입 추가 #110

ujiiin committed:
[BUGFIX] guestUser status constant 추가  #110

ujiiin committed:
[BUGFIX] 멤버 카운트 함수 추가 #110

ujiiin committed:
[BUGFIX] 멤버 함수 매칭내역 API에 적용 #110

ujiiin committed:
[BUGFIX] main branch pull 중 에러 해결 #103

ujiiin committed:
[BUGFIX] 게스팅 목록 조회 및 신청 에러 해결 #103

ujiiin committed:
[BUGFIX] main branch pull 중 충돌 해결 #110

ujiiin committed:
[BUGFIX] 매칭내역 조회 에러 해결 #110

mzeong committed:
Fix: 게스트 모집글 작성, 수정, 상세 조회, 신청은 로그인된 유저만 가능하도록 수정 #110

mzeong committed:
Fix: 매칭내역 페이지 내 모든 API는 로그인된 유저만 가능하도록 수정 #110

mzeong committed:
Fix: guests.controller.ts 파일 수정 #110

mzeong commented on src/daos/team.dao.ts:
```diff
@@ -72,7 +72,7 @@ export const getTeamDetail = async (teamId: number) => {
 
 export const getTeamDetailforGuesting = async (teamId: number) => {
-    return await db.Team.findOne({
+    return await db.Team.findAll({
```
코드를 이렇게 변경하게 되면 기존에 getTeamDetailForGuesting 함수를 사용하고 있는 다른 코드 부분에서 에러가 나게 됩니다 
+이 부분은 제가 수정해서 커밋했습니다

mzeong committed:
Fix: getTeamDetailForGuesting 함수 수정 #110

mzeong committed:
Fix: findMemberInfoByCategory 함수를 통한 멤버 프로필 조회 #110

mzeong committed:
Style: insertGuestUser로 함수 이름 변경 #110

mzeong committed:
Style: 더 이상 사용하지 않는 findMemberInfoByTeamId 함수 제거 #110

mzeong committed:
Comment: addMemberCount 함수에 경고 메세지 추가 #110

mzeong commented on src/services/matchings.service.ts:
```diff
@@ -1,60 +1,59 @@
-    for (const matchingGame of matchingGames) {
-        matchingGame.memberCount = (await getMemberCountByTeamId(matchingGame["Team.id"])) + 1;
-    }
+    const teamId = await getTeamIdByLeaderId(userId);
```
유저가 여러 팀의 리더가 될 수 있기 때문에 이 부분은 수정이 필요합니다

ujiiin committed:
Fix: 유저(팀장)의 모든 카테고리 게스팅, 호스팅 내역 조회 가능하도록 수정 #110

ujiiin committed:
Fix: 게스트 모집글 수정 API에서 teamId 가져오는 부분 수정 #110

ujiiin commented on src/daos/member.dao.ts:
```diff
@@ -98,6 +98,7 @@ export const getMemberCountByTeamId = async (teamId) => {
 };
 
 export const addMemberCount = async (lists) => {
+    console.log("addMemberCount 함수는 제거될 함수입니다");
```
이 부분은 왜 추가된걸까요?

mzeong replied:
추후 최적화가 되어야 할 함수라서 적어놨습니다. 나중에 일괄로 교체할거라 일단은 계속 사용하셔도 됩니다.

mzeong committed:
Fix: 날짜별 호스팅 내역 조회 API 수정 #110

- 유저가 팀장인 팀이 게스트를 모집한 경기, 상대팀을 모집한 경기를 조회하는 API
- 유저가 팀장인 팀을 모두 조회하도록 수정
- findGamesOfMatchingHosting 함수에서 game, team 테이블 간 조회가 안 되는 문제 해결
- findGuestsOfMatchingHosting, findGamesOfMatchingHosting 함수 이름 변경 및 파일 이동
- readMatchingHostingResponseDTO 내에서 경기시간 순 정렬
- 경기 타입을 이넘으로 관리

mzeong committed:
Merge branch 'bugfix/110' of https://github.com/TiimMate/TeamMateServer into bugfix/110

mzeong committed:
Fix: 날짜별 게스팅 내역 조회 API 수정 #110

- 유저가 게스트로 신청한 경기, 유저가 팀장인 팀이 상대팀으로 신청한 경기를 조회하는 API
